src/toasts/active-toasts.component.js

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import { toastsSubject } from "./toasts";
+import Toast from "./toast.component";
+
+export default function ActiveToasts() {
+  const [toasts, setToasts] = React.useState([]);
+  const [toastsClosing, setToastsClosing] = React.useState([]);
+  const closeToastRef = React.useRef();
+  closeToastRef.current = closeToast;
+
+  React.useEffect(() => {
+    const subscription = toastsSubject.subscribe(toast =>
+      setToasts([...toasts, toast])
+    );
+
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, [toastsSubject, toasts]);
+
+  React.useEffect(() => {
```

Comment:
This is ok with the framework agnostic toast or the requirements was changed?

Reply:
The `showToast()` function is just a function that any framework can call. The internal implementation happens to use React, but the user of `showToast()` doesn't know that and doesn't need to be using react.